bokehjs/src/coffee/renderer/tile/tile_renderer.coffee

```diff
@@ -0,0 +1,215 @@
+_ = require "underscore"
+HasParent = require "../../common/has_parent"
+PlotWidget = require "../../common/plot_widget"
+properties = require "../../common/properties"
+wmts = require "./wmts_tile_source"
+ImagePool = require "./image_pool"
+{logger} = require "../../common/logging"
+
+class TileRendererView extends PlotWidget
+  
+  bind_bokeh_events: () ->
+    @listenTo(@model, 'change', @request_render)
+
+  get_extent: () ->
+    return [@x_range.get('start'), @y_range.get('start'), @x_range.get('end'), @y_range.get('end')]
+
+  _set_data: () ->
+    @pool = new ImagePool()
+    @map_plot = @plot_view.model
+    @map_canvas = @plot_view.canvas_view.ctx
+    @map_frame = @plot_view.frame
+    @x_range = @map_plot.get('x_range')
+    @x_mapper = this.map_frame.get('x_mappers')['default']
+    @y_range = @map_plot.get('y_range')
+    @y_mapper = this.map_frame.get('y_mappers')['default']
+
+  _map_data: () ->
+    @initial_extent = @get_extent()
+    console.dir(@mget('tile_source'))
+    console.warn(@mget('tile_source'))
+    console.warn(@mget('tile_source'))
+    zoom_level = @mget('tile_source').get_level_by_extent(@initial_extent, @map_frame.get('height'), @map_frame.get('width'))
+    new_extent = @mget('tile_source').snap_to_zoom(@initial_extent, @map_frame.get('height'), @map_frame.get('width'), zoom_level)
+    @x_range.set('start', new_extent[0])
+    @y_range.set('start', new_extent[1])
+    @x_range.set('end', new_extent[2])
+    @y_range.set('end', new_extent[3])
+
+  _on_tile_load: (e) =>
+    tile_data = e.target.tile_data
+    tile_data.img = e.target
+    tile_data.current = true
+    @mget('tile_source').tiles[tile_data.cache_key] = tile_data
+    @_render_tile(tile_data.cache_key)
```

Comment:
So my current thought is that its my use of `setTimeout` in `render()` which serves to prevent unnecessary tile requests during mouse-wheel zooming.  My guess is that the render cycle completes but then 65ms later I render some tiles. Thinking of ways to refactor this now...